manualsource/src/main/java/org/hibernate/search/manualsource/source/ObjectInitializer.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+
+package org.hibernate.search.manualsource.source;
+
+import java.util.List;
+import java.util.Map;
+
+import org.hibernate.search.query.engine.spi.TimeoutManager;
+
+/**
+ * Interface implemented by the source of data to load specific entities.
+ *
+ * @author Emmanuel Bernard &lt;emmanuel@hibernate.org&gt;
+ */
+public interface ObjectInitializer {
+	/**
+	 * Initializes the set of requested objects and put them in the map.
+	 */
+	void initializeObjects(List<EntityKeyForLoad> keys, Map<EntityKeyForLoad, Object> idsToObjects, Context context);
```

Comment:
Out of interest, what's the reason for passing in the result map as a parameter instead of returning it?

A pattern growing on me recently is passing in a "collector" accepting result(s):

    void initializeObjects(List<EntityKeyForLoad> keys, EntityCollector collector, Context context) {
        for(EntityKeyForLoad key : keys) {
            Object entity = ...;
            collector.addEntity( key, entity );
        }
    }

Essentially, it's a more abstract flavor of yours, exposing only those methods actually needed. Plus, it gives our caller the freedom of how to implement it and we can add more methods to that down the road if needed: